File: Sport_Backend/src/main/java/com/example/sport_backend/Entity/Health/RecoveryPlan.java
package com.example.sport_backend.Entity.Health;

import com.example.sport_backend.Entity.ClubHouse.Player;
import com.example.sport_backend.Entity.Enum.PlanStatus;
import com.example.sport_backend.Entity.Enum.PlanType;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity

public class RecoveryPlan {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String planDescription;
    LocalDate startDate;
    LocalDate estimatedEndDate;
    LocalDate actualEndDate;

    Float progress;
    Integer sessionFrequency;
    Integer sessionDuration;

    @Enumerated(EnumType.STRING)
    PlanType planType;

    LocalDate nextReviewDate;
    String adjustments;

    @Enumerated(EnumType.STRING)
    PlanStatus planStatus;

    @OneToOne
    @JoinColumn(name = "injury_id")
    private Injury injury;

    @ManyToOne
    @JoinColumn(name = "player_id", nullable = false)
    private Player player; // Association avec Player


    // Le playerId est juste utilisé pour la création et n'est pas persisté dans la DB
    @Transient
    private Long playerId;




}

File: Sport_FrontEnd/SportManagment/src/app/Health/models/recoveryplan.ts
import { Injury } from './injury';

export interface RecoveryPlan {
  id: number;
  planDescription: string;
  startDate: string;            // LocalDate donc string ISO (ex: '2025-02-27')
  estimatedEndDate: string;     // idem
  actualEndDate?: string | null; // Optionnel, car pas toujours renseigné
  progress: number;              // Float
  sessionFrequency: number;
  sessionDuration: number;
  planType: PlanType;
  nextReviewDate: string;
  adjustments: string;
  planStatus: PlanStatus;
  injury: Injury;


}


// Enums associés (frontend Angular)
export enum PlanType {
  PHYSIOTHERAPY = 'PHYSIOTHERAPY',
  REST = 'REST',
  STRENGTH_TRAINING = 'STRENGTH_TRAINING'
}


export enum PlanStatus {
  EN_COURS = 'EN_COURS',
  TERMINE = 'TERMINE',
  SUSPENDU = 'SUSPENDU'
}

